cli/tests/testdata/npm/compare_globals/main.ts

```diff
@@ -12,3 +12,16 @@ type _TestHasNodeJsGlobal = NodeJS.Architecture;
 
 const controller = new AbortController();
 controller.abort("reason"); // in the NodeJS declaration it doesn't have a reason
+
+// Super edge case where some Node code deletes a global where the
+// Node code has its own global and the Deno code has the same global,
+// but it's different. Basically if some Node code deletes
+// one of these globals then we don't want it to suddenly inherit
+// the Deno global.
+globals.withNodeGlobalThis((nodeGlobalThis: any) => {
+  (globalThis as any).setTimeout = 5;
+  console.log(setTimeout);
+  delete nodeGlobalThis["setTimeout"];
+  console.log(nodeGlobalThis["setTimeout"]); // should be undefined
+  console.log(globalThis["setTimeout"]); // should be undefined
```

Comment:
On main these both go to `undefined`. I'm not entirely sure if this one should, but that's the current behaviour